multiversx_sdk/core/base_controller.py

```diff
@@ -37,9 +40,14 @@ def _set_transaction_gas_options(
 
         if gas_limit:
             transaction.gas_limit = gas_limit
+            return
         else:
             self._add_extra_gas_limit_if_required(transaction)
 
+        if self.gas_limit_estimator:
+            transaction.gas_limit = 0
```

Comment:
Maybe add a comment which says "we have to reset ... because ...". On the other hand, isn't the estimator itself doing that (on a transaction clone)?

Reply:
added